feature-dapp-impl/src/main/java/io/novafoundation/nova/feature_dapp_impl/utils/tabs/models/BrowserTabSession.kt

```diff
@@ -4,67 +4,60 @@ import android.content.Context
 import android.content.Intent
 import android.graphics.Bitmap
 import android.webkit.WebView
+import io.novafoundation.nova.common.resources.ContextManager
 import io.novafoundation.nova.feature_dapp_impl.web3.webview.PageCallback
 import io.novafoundation.nova.feature_dapp_impl.web3.webview.Web3ChromeClient
+import io.novafoundation.nova.feature_dapp_impl.web3.webview.CompoundWeb3Injector
 import io.novafoundation.nova.feature_dapp_impl.web3.webview.Web3WebViewClient
 import io.novafoundation.nova.feature_dapp_impl.web3.webview.injectWeb3
 import io.novafoundation.nova.feature_dapp_impl.web3.webview.uninjectWeb3
-import java.util.Date
 
-class PageSessionFactory {
+class BrowserTabSessionFactory(
+    private val contextManager: ContextManager,
+    private val compoundWeb3Injector: CompoundWeb3Injector
+) {
 
-    fun create(tabId: String, sessionStartTime: Date, startUrl: String): PageSession {
-        return PageSession(
+    fun create(tabId: String, startUrl: String): BrowserTabSession {
+        return BrowserTabSession(
             tabId = tabId,
-            sessionStartTime = sessionStartTime,
-            startUrl = startUrl
+            startUrl = startUrl,
+            context = contextManager.getActivity()!!,
+            compoundWeb3Injector = compoundWeb3Injector
         )
     }
 }
 
-class PageSession(
+class BrowserTabSession(
     val tabId: String,
-    val sessionStartTime: Date,
-    val startUrl: String
+    val startUrl: String,
+    private val context: Context,
+    private val compoundWeb3Injector: CompoundWeb3Injector
 ) : PageCallback {
 
     private var _webView: WebView? = null
-    val webView: WebView
-        get() {
-            return _webView ?: throw IllegalStateException("WebView is not initialized")
-        }
+    val webView: WebView by lazy {
+        initialize()
+        _webView!!.loadUrl(startUrl)
+        _webView!!
+    }
```

Comment:
It seems like the whole nullability and `initialize` workarounds are here just to initialize  `_webView` and `_webViewClient`  at the same time, when accessing either of them. Which is too implicit, like why would I expect one field to be initialized when accessing another? I suggest to do it in more explicit way:

``` kotlin
class SessionEnviroment(val webView: WebView, val client: Web3WebViewClient)

val enviroment: SessionEnviroment by lazy {
  createSessionEnviroment()
}
```

Reply:
Solve it in previous pr.
Since I pass webView via constructor it's not necessary to create a SessionEnviroment